src/Integration/LocalServices/SolutionBindingInformationProvider.cs

```diff
@@ -31,6 +31,7 @@ namespace SonarLint.VisualStudio.Integration
     internal class SolutionBindingInformationProvider : ISolutionBindingInformationProvider
     {
         private readonly IServiceProvider serviceProvider;
+        private readonly ISolutionRuleSetsInformationProvider ruleSetInfoProvider;
```

Comment:
Simplification: the _ruleSetInfoProvider_ service was fetched from three separate places in the code, and was also re-fetched for each project when looping through the list of projects. I've changed it to be fetched once and stored.